### ISSUE TITLE
IOS Safari Lockscreen

### ISSUE THREAD

**Issue by Fuchsoria** [Tonejs/Tone.js#995]
**Describe the bug**

Audio stops after screenlock in ios 14+

**To Reproduce**
Open https://tonejs.github.io/examples/player in safari/chrome on ios 15+ and lock the screen, then it's automaticly stops audio

**Expected behavior**
Audio continue to play when the screen is locked

**What I've tried**
On ios 12 in safari it's working, but in ios 14/15 isn't working anymore

**Additional context**
I'm using `next` version of npm package

**Comment by estafaa** [comment]
Same issue with also p5.js library. sound stops is the screen is locked or safari minimazed. If open new tab in the browser and then lock the screen then it continues playing. 
issues with iOS 15+

**Comment by estafaa** [comment]
Hi @snuffyDev , I think this issue is at the WebKit level (because all browsers on iOS 15 are affected, Safari, Chrome, Firefox ert.). In one of our projects we use Tone.js and in another p5.js and in both cases we have the same issue. I added a video demonstrating this bug here: https://github.com/processing/p5.js-sound/issues/648

same behavior if Tone.js is used. 

If you experiencing the same, please check this case for Apple support and maybe add your comment to increase the priority of the issue. https://feedbackassistant.apple.com/feedback/9709046

Thank you!

**Comment by tambien** [comment]
Thanks for the info @estafaa 

I agree, unfortunately it doesn't seem like there's anything we can do in Tone.js to change this if it's not supported in the latest iOS

**Comment by pendragon-andyh** [comment]
Recent posting on [Webaudio slck channel](https://web-audio.slack.com/archives/C04PVA2DH/p1640876727146000) claims to work around the issue.

**Comment by tambien** [comment]
Thanks @pendragon-andyh could you summarize the response here for people who aren't on Slack?

**Comment by pendragon-andyh** [comment]
Pasted from Slack:
> ... this was sufficient to fix it:
```
 if (context.state === 'suspended' || context.state === 'interrupted') {
   context.resume()
 }
 // I put the above in the audio ctx onstatechange handler
```
> I still need the html5 autoplaying silent keep alive file to be able to have the web audio work by default when the phone is in silent mode, 
but otherwise working…

>

**Comment by estafaa** [comment]
Yes, it works! Amazing, thank you sooo much! 

> Pasted from Slack:
> 
> > ... this was sufficient to fix it:
> 
> ```
>  if (context.state === 'suspended' || context.state === 'interrupted') {
>    context.resume()
>  }
>  // I put the above in the audio ctx onstatechange handler
> ```
> 
>  > I still need the html5 autoplaying silent keep alive file to be able to have the web audio work by default when the phone is in silent mode,
> > but otherwise working…
